e2e/tests/app.e2e.ts

```diff
@@ -0,0 +1,148 @@
+
+      await angularMaterialTabsPage.clickOnButtonsTab();
+      await expect(angularMaterialTabsPage.buttons).toHaveText('Buttons');
+      await expect(angularMaterialTabsPage.tabThreeContent).not.toBeVisible();
+    });
+
+    test('Buttons', async ({page}) => {
+      const angularMaterialButtonsPage = new AngularMaterialButtonsPage(page);
+
+      await page.waitForLoadState('domcontentloaded');
+
+      await expect(await angularMaterialButtonsPage.basicButtons.all()).toHaveLength(5);
+      await expect(await angularMaterialButtonsPage.basicLinks.all()).toHaveLength(1);
+      await expect(angularMaterialButtonsPage.disabledBasicButtons).toBeDisabled();
+
+      await expect(await angularMaterialButtonsPage.raisedButtons.all()).toHaveLength(5);
+      await expect(await angularMaterialButtonsPage.raisedLinks.all()).toHaveLength(1);
+      await expect(angularMaterialButtonsPage.disabledRaisedButtons).toBeDisabled();
+
+      await expect(await angularMaterialButtonsPage.strokedButtons.all()).toHaveLength(5);
+      await expect(await angularMaterialButtonsPage.strokedLinks.all()).toHaveLength(1);
+      await expect(angularMaterialButtonsPage.disabledStrokedButtons).toBeDisabled();
+
+      await expect(await angularMaterialButtonsPage.flatButtons.all()).toHaveLength(5);
+      await expect(await angularMaterialButtonsPage.flatLinks.all()).toHaveLength(1);
+      await expect(angularMaterialButtonsPage.disabledFlatButtons).toBeDisabled();
+
+      await expect(await angularMaterialButtonsPage.iconButtons.all()).toHaveLength(5);
+    });
+
+    test('CDK Tree', async ({page}) => {
+      const angularMaterialTabsPage = new AngularMaterialTabsPage(page);
+      const angularMaterialCdkTreePage = new AngularMaterialCdkTreePage(page);
+
+      await angularMaterialTabsPage.clickOnCdkTreeTab();
+
+      await angularMaterialCdkTreePage.clickOnFruit();
+      await expect(angularMaterialCdkTreePage.apple).toBeVisible();
+
+      await angularMaterialCdkTreePage.clickOnFruit();
+      await expect(angularMaterialCdkTreePage.apple).not.toBeVisible();
+
+      await angularMaterialCdkTreePage.clickOnVegetables();
+      await angularMaterialCdkTreePage.clickOnGreen();
+      await expect(angularMaterialCdkTreePage.broccoli).toBeVisible();
+
+      await angularMaterialCdkTreePage.clickOnOrange();
+      await expect(angularMaterialCdkTreePage.pumpkins).toBeVisible();
+
+      await angularMaterialCdkTreePage.clickOnVegetables();
+      await expect(angularMaterialCdkTreePage.broccoli).not.toBeVisible();
+      await expect(angularMaterialCdkTreePage.pumpkins).not.toBeVisible();
+
+    });
+  });
+
+  test.describe('TopNav', () => {
+    test('ToNav title', async ({page}) => {
+      const topNavPage = new TopNavPage(page);
+      const title = await topNavPage.title;
+      await expect(title).toHaveText('Angular Upgrade and Angular CLI Demo');
+    });
+
+    test('Redirect to GitHub page', async ({page}) => {
```

Comment:
"Redirect" does not sound accurate. (It does not redirect to GitHub, it open a new tab.)
Maybe change to something like:

```suggestion
    test('GitHub link', async ({page}) => {
```

Reply:
Changed